tutorial/openmp.md

```diff
@@ -0,0 +1,583 @@
+❯ pyccel omp_test.py --openmp
+❯ ./omp_test
+893116
+```
+
+### Single Construct
+
+#### Syntax
+
+```python
+#$ omp single [clause[ [,] clause] ... ]
+structured-block
+#$ omp end single [end_clause[ [,] end_clause] ... ]
+```
+
+#### Example
+
+This example shows how we can use the ``` #$ omp single ``` pragma to specify a section of code that must be run by a single available thread.
+
+```python
+from pyccel.stdlib.internal.openmp import omp_set_num_threads, omp_get_num_threads, omp_get_thread_num
+omp_set_num_threads(4)
+#$ omp parallel
+print("hello from thread number:", omp_get_thread_num())
+#$ omp single
+print("The best thread is number : ", omp_get_thread_num())
+#$ omp end single
+#$ omp end parallel
+```
+
+The output of this program is (you may get different result because of threads running at the same time):
+```shell
+❯ pyccel omp_test.py --openmp
+❯ ./omp_test
+hello from thread number:            1
+The best thread is number :             1
+hello from thread number:            2
+hello from thread number:            3
+hello from thread number:            0
+```
+
+### Critical Construct
+
+#### Syntax
+
+```python
+#$ omp critical [(name) [ [,] hint (hint-expression)]]
+structured-block
+#$ omp end critical
+```
+
+#### Example
+
+This example shows how ``` #$ omp critical ``` is used to specify the code which must be executed by one thread at a time.
+```python
+sum = 0
+#$ omp parallel num_threads(4) private(i) shared(sum)
+#$ omp for
+for i in range(0, 1337):
+  #$ omp critical
+  sum += i
+  #$ omp end critical
+#$ omp end parallel
+print(sum)
+```
+
+The output of this program is:
+```shell
+❯ pyccel omp_test.py --openmp
+❯ ./omp_test
+893116
+```
+
+### Barrier Construct
+
+#### Syntax
+
+```python
+#$ omp barrier
+```
+
+#### Example
+
+This example shows how ``` #$ omp barrier ``` is used to specify a point in the code where each thread must wait until all threads in the team arrive.
+```python
+from numpy import zeros
+
+n = 1337
+arr = zeros((n))
+arr_2 = zeros((n))
+#$ omp parallel num_threads(4) private(i, j) shared(arr)
+
+#$ omp for
+for i in range(0, n):
+  arr[i] = i
+#$ omp barrier
+#$ omp for
+for j in range(0, n):
+  arr_2[j] = arr[j] * 2
+
+#$ omp end parallel
+print(sum(arr_2))
+```
+
+The output of this program is:
+```shell
+❯ pyccel omp_test.py --openmp
+❯ ./omp_test
+1786232
+```
+
+### Masked Construct
+
+#### Syntax
+
+```python
+#$ omp masked [ filter(integer-expression) ]
+structured-block
+#$ omp end masked
+```
+
+#### Example
+
+The ``` #$ omp masked ``` pragma is used here to specify a structured block that is executed by a subset of the threads of the current team.
+```python
+result = 0
+#$ omp parallel num_threads(4)
+#$ omp masked
+result = result + 1
+#$ omp end masked
+#$ omp end parallel
+print("result :", result)
+```
+
+The output of this program is:
+```shell
+❯ pyccel omp_test.py --openmp
+❯ ./omp_test
+result : 1
+```
+
+### Taskloop/Atomic Construct
+
+#### Syntax Taskloop
```

Comment:
I think that this should read

`#### Syntax of *taskloop*`

which is displayed as

#### Syntax of *taskloop*

I suggest you use the same phrasing and formatting throughout the document.